guide/interactions/buttons.md

```diff
@@ -0,0 +1,321 @@
+
+	if (interaction.commandName === 'ping') {
+		const row = new MessageActionRow()
+			.addComponents(
+				new MessageButton()
+					.setCustomID('primary')
+					.setLabel('primary')
+					.setStyle('PRIMARY')
+					.setEmoji('123456789012345678'),
+			);
+
+		await interaction.reply({ content: 'Pong!', components: [row] });
+	}
+});
+```
+
+Now you know all there is to building and sending a `MessageButton`! Let's move on to how to receive buttons!
+
+
+## Receiving buttons
+
+Whilst you can receive and handle a `ButtonInteraction` via the interaction event, we recommend you use one of the collectors we'll be covering in the next section. To receive a `ButtonInteraction`, attach an event listener to your client and use the `Interaction#isButton()` type guard to make sure you only receive component interactions.
+
+```js {2}
+client.on('interaction', interaction => {
+	if (!interaction.isButton()) return;
+	console.log(interaction);
+});
+```
+
+
+## Component collectors
+
+These work quite similarly to message and reaction collectors, except that you will receive instances of the `MessageComponentInteraction` class as collected items.
+
+::: tip
+You can create the collectors on either a `message` or a `channel`.
+:::
+
+### MessageComponentInteractionCollector
+
+Here's how you can create a basic event-based `MessageComponentInteractionCollector`:
+
+```js {6,8,10-11}
+client.on('interaction', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		const message = await interaction.fetchReply();
+		const filter = i => i.customID === 'primary' && i.user.id === '122157285790187530';
+
+		const collector = message.createMessageComponentInteractionCollector(filter, { time: 15000 });
+
+		collector.on('collect', i => console.log(`Collected ${i.customID}`));
+		collector.on('end', collected => console.log(`Collected ${collected.size} items`));
+	}
+});
+```
+
+::: danger
+For ephemeral responses you cannot fetch a message object, so create the collector on a channel instead.
+:::
+
+### awaitMessageComponentInteraction
+
+As with other types of collectors, you can also use a promise-based collector like this:
+
+::: warning
+Unlike other promise-based collectors, this one only collects a single item!
+:::
+
+```js 8-11}
+client.on('interaction', async interaction => {
+	if (!interaction.isCommand()) return;
+
+	if (interaction.commandName === 'ping') {
+		const message = await interaction.fetchReply();
+		const filter = i => i.customID === 'primary' && i.user.id === '122157285790187530';
+
+		message.awaitMessageComponentInteraction(filter, { time: 15000 })
+			.then(i => console.log(`${i.customID} was clicked!`))
+			.catch(console.error);
+	}
+});
+```
+
+
+## Responding to buttons
+
+The `MessageComponentInteraction` class has similar methods as the `CommandInteraction` class. These methods behave exactly the same as on the `CommandInteraction` class:
```

Comment:
similar or exactly the same? decide on one, remove the word doubling, if possible.